src/extension.ts

```diff
@@ -99,14 +100,21 @@ function onDidChangeTelemetryEnabled(): Disposable {
 
 async function openTelemetryOptInDialogIfNeeded(context: ExtensionContext) {
   logTelemetryStatus();
-  const optInRequested: boolean | undefined = context.globalState.get(
-    OPT_IN_STATUS_KEY
-  );
+  const optInRequested: boolean | undefined = context.globalState.get(OPT_IN_STATUS_KEY);
+
   Logger.log(`optInRequested is: ${optInRequested}`);
 
   if (optInRequested) {
+    if (retryOptin) {
+      clearInterval(retryOptin);
+    }
```

Comment:
is this required? you would already cleared when use opts in from the popup. #line 115